From a23fdda08a4f34b815bc2e02e7c7241b82780a68 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Thu, 22 Oct 2020 09:30:54 +0100
Subject: [PATCH] testsuite, Darwin, Arm64 : Skip more tests using
 'check-function-bodies'.

The implementation for check-function-bodies doesn't support Mach-O at this
point - so this is justy noise.  TODO: provide a Mach-O version of the scan
asm function.

(cherry picked from commit 8bd52be95e400e46f3644f4e741e80897c570d81)
Signed-off-by: Kirill A. Korinsky <kirill@korins.ky>
---
 .../gcc.target/aarch64/advsimd-intrinsics/bfcvt-compile.c       | 2 +-
 .../gcc.target/aarch64/advsimd-intrinsics/bfcvt-nosimd.c        | 2 +-
 .../gcc.target/aarch64/advsimd-intrinsics/bfcvtnq2-untied.c     | 2 +-
 gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-1.c   | 2 +-
 gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-2.c   | 2 +-
 .../gcc.target/aarch64/advsimd-intrinsics/bfmlalbt-compile.c    | 2 +-
 .../gcc.target/aarch64/advsimd-intrinsics/bfmmla-compile.c      | 2 +-
 gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-1.c  | 2 +-
 gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-2.c  | 2 +-
 gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_1.c            | 2 +-
 gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_2.c            | 2 +-
 gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_3.c            | 2 +-
 gcc/testsuite/gcc.target/aarch64/bfloat16_simd_1.c              | 2 +-
 gcc/testsuite/gcc.target/aarch64/bfloat16_simd_2.c              | 2 +-
 gcc/testsuite/gcc.target/aarch64/bfloat16_simd_3.c              | 2 +-
 gcc/testsuite/gcc.target/aarch64/movk_2.c                       | 2 +-
 gcc/testsuite/gcc.target/aarch64/pr92424-1.c                    | 2 +-
 gcc/testsuite/gcc.target/aarch64/pr94697.c                      | 2 +-
 gcc/testsuite/gcc.target/aarch64/torture/simd-abi-9.c           | 2 +-
 gcc/testsuite/gcc.target/aarch64/vect_mixed_sizes_14.c          | 2 +-
 20 files changed, 20 insertions(+), 20 deletions(-)

diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvt-compile.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvt-compile.c
index 47af7c494d9..e4175c077a4 100644
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvt-compile.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvt-compile.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-add-options arm_v8_2a_bf16_neon } */
 /* { dg-additional-options "-save-temps" } */
-/* { dg-final { check-function-bodies "**" "" {-O[^0]} } } */
+/* { dg-final { check-function-bodies "**" "" {-O[^0]} { target { ! aarch64*-*-darwin* } } } } */
 /* { dg-skip-if "" { *-*-* } { "-fno-fat-lto-objects" } } */
 
 #include <arm_neon.h>
diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvt-nosimd.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvt-nosimd.c
index a914680937d..c6b2ef3e444 100644
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvt-nosimd.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvt-nosimd.c
@@ -2,7 +2,7 @@
 /* { dg-skip-if "" { *-*-* } { "-fno-fat-lto-objects" } } */
 /* { dg-require-effective-target aarch64_asm_bf16_ok } */
 /* { dg-additional-options "-save-temps -march=armv8.2-a+bf16+nosimd" } */
-/* { dg-final { check-function-bodies "**" "" {-O[^0]} } } */
+/* { dg-final { check-function-bodies "**" "" {-O[^0]} { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_neon.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvtnq2-untied.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvtnq2-untied.c
index 4b730e39d4e..e0795394c6e 100644
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvtnq2-untied.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfcvtnq2-untied.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-add-options arm_v8_2a_bf16_neon } */
 /* { dg-additional-options "-save-temps" } */
-/* { dg-final { check-function-bodies "**" "" {-O[^0]} } } */
+/* { dg-final { check-function-bodies "**" "" {-O[^0]} { target { ! aarch64*-*-darwin* } } } } */
 /* { dg-skip-if "" { *-*-* } { "-fno-fat-lto-objects" } } */
 
 #include <arm_neon.h>
diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-1.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-1.c
index ad51507731b..cf3a66a4fb4 100755
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-1.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-1.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-add-options arm_v8_2a_bf16_neon }  */
 /* { dg-additional-options "-save-temps" } */
-/* { dg-final { check-function-bodies "**" "" {-O[^0]} } } */
+/* { dg-final { check-function-bodies "**" "" {-O[^0]} { target { ! aarch64*-*-darwin* } } } } */
 /* { dg-skip-if "" { *-*-* } { "-fno-fat-lto-objects" } } */
 
 #include <arm_neon.h>
diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-2.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-2.c
index 58bdee5ac9d..055cf62b78a 100755
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-2.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfdot-2.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-add-options arm_v8_2a_bf16_neon }  */
 /* { dg-additional-options "-mbig-endian --save-temps" } */
-/* { dg-final { check-function-bodies "**" "" {-O[^0]} } } */
+/* { dg-final { check-function-bodies "**" "" {-O[^0]} { target { ! aarch64*-*-darwin* } } } } */
 /* { dg-skip-if "" { *-*-* } { "-fno-fat-lto-objects" } } */
 
 #include <arm_neon.h>
diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfmlalbt-compile.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfmlalbt-compile.c
index 9810e4ba374..37baea2fb05 100644
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfmlalbt-compile.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfmlalbt-compile.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-add-options arm_v8_2a_bf16_neon } */
 /* { dg-additional-options "-save-temps" } */
-/* { dg-final { check-function-bodies "**" "" "-DCHECK_ASM" } } */
+/* { dg-final { check-function-bodies "**" "" "-DCHECK_ASM" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_neon.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfmmla-compile.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfmmla-compile.c
index 0aaa69f0037..fdd69b52d67 100644
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfmmla-compile.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/bfmmla-compile.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-add-options arm_v8_2a_bf16_neon } */
 /* { dg-additional-options "-save-temps" } */
-/* { dg-final { check-function-bodies "**" "" "-DCHECK_ASM" } } */
+/* { dg-final { check-function-bodies "**" "" "-DCHECK_ASM" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_neon.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-1.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-1.c
index ac4f821e771..f20608de05c 100755
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-1.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-1.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_i8mm_ok } */
 /* { dg-add-options arm_v8_2a_i8mm }  */
 /* { dg-additional-options "-save-temps" } */
-/* { dg-final { check-function-bodies "**" "" {-O[^0]} } } */
+/* { dg-final { check-function-bodies "**" "" {-O[^0]} { target { ! aarch64*-*-darwin* } } } } */
 /* { dg-skip-if "" { *-*-* } { "-fno-fat-lto-objects" } } */
 
 #include <arm_neon.h>
diff --git gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-2.c gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-2.c
index 96bca2356e4..f833486b6d9 100755
--- gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-2.c
+++ gcc/testsuite/gcc.target/aarch64/advsimd-intrinsics/vdot-3-2.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_i8mm_ok } */
 /* { dg-add-options arm_v8_2a_i8mm }  */
 /* { dg-additional-options "-mbig-endian -save-temps" } */
-/* { dg-final { check-function-bodies "**" "" {-O[^0]} } } */
+/* { dg-final { check-function-bodies "**" "" {-O[^0]} { target { ! aarch64*-*-darwin* } } } } */
 /* { dg-skip-if "" { *-*-* } { "-fno-fat-lto-objects" } } */
 
 #include <arm_neon.h>
diff --git gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_1.c gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_1.c
index ef43766495c..ff2801ddaa5 100644
--- gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_1.c
+++ gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_1.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-add-options arm_v8_2a_bf16_neon }  */
 /* { dg-additional-options "-O3 --save-temps -std=gnu90" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_bf16.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_2.c gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_2.c
index df8e7518c24..6c0f0275345 100644
--- gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_2.c
+++ gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_2.c
@@ -1,7 +1,7 @@
 /* { dg-do assemble { target { aarch64*-*-* } } } */
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-additional-options "-march=armv8.2-a -O3 --save-temps -std=gnu90" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_bf16.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_3.c gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_3.c
index 5d7a4317cee..83517a450cf 100644
--- gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_3.c
+++ gcc/testsuite/gcc.target/aarch64/bfloat16_scalar_3.c
@@ -1,7 +1,7 @@
 /* { dg-do assemble { target { aarch64*-*-* } } } */
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-additional-options "-march=armv8.2-a -O3 --save-temps -std=gnu90" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_bf16.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/bfloat16_simd_1.c gcc/testsuite/gcc.target/aarch64/bfloat16_simd_1.c
index 6cad557ebf2..b9bddb368f0 100644
--- gcc/testsuite/gcc.target/aarch64/bfloat16_simd_1.c
+++ gcc/testsuite/gcc.target/aarch64/bfloat16_simd_1.c
@@ -2,7 +2,7 @@
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-add-options arm_v8_2a_bf16_neon }  */
 /* { dg-additional-options "-O3 --save-temps" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_neon.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/bfloat16_simd_2.c gcc/testsuite/gcc.target/aarch64/bfloat16_simd_2.c
index 3891dcfc900..73e5ec9d158 100644
--- gcc/testsuite/gcc.target/aarch64/bfloat16_simd_2.c
+++ gcc/testsuite/gcc.target/aarch64/bfloat16_simd_2.c
@@ -1,7 +1,7 @@
 /* { dg-do assemble { target { aarch64*-*-* } } } */
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-additional-options "-march=armv8.2-a -O3 --save-temps" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_neon.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/bfloat16_simd_3.c gcc/testsuite/gcc.target/aarch64/bfloat16_simd_3.c
index b35f5e527be..0737f0d4060 100644
--- gcc/testsuite/gcc.target/aarch64/bfloat16_simd_3.c
+++ gcc/testsuite/gcc.target/aarch64/bfloat16_simd_3.c
@@ -1,7 +1,7 @@
 /* { dg-do assemble { target { aarch64*-*-* } } } */
 /* { dg-require-effective-target arm_v8_2a_bf16_neon_ok } */
 /* { dg-additional-options "-march=armv8.2-a -O3 --save-temps" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <arm_neon.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/movk_2.c gcc/testsuite/gcc.target/aarch64/movk_2.c
index a0477ad5d42..43420b5a06b 100644
--- gcc/testsuite/gcc.target/aarch64/movk_2.c
+++ gcc/testsuite/gcc.target/aarch64/movk_2.c
@@ -1,6 +1,6 @@
 /* { dg-do compile } */
 /* { dg-options "-O" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 #include <stdint.h>
 
diff --git gcc/testsuite/gcc.target/aarch64/pr92424-1.c gcc/testsuite/gcc.target/aarch64/pr92424-1.c
index c413a2c306e..995b48eeb9f 100644
--- gcc/testsuite/gcc.target/aarch64/pr92424-1.c
+++ gcc/testsuite/gcc.target/aarch64/pr92424-1.c
@@ -1,6 +1,6 @@
 /* { dg-do "compile" } */
 /* { dg-options "-O1" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 /* Note: this test only checks the instructions in the function bodies,
    not the placement of the patch label or nops before the futncion.  */
diff --git gcc/testsuite/gcc.target/aarch64/pr94697.c gcc/testsuite/gcc.target/aarch64/pr94697.c
index fcc9ab87f15..36e9cfdaa2e 100644
--- gcc/testsuite/gcc.target/aarch64/pr94697.c
+++ gcc/testsuite/gcc.target/aarch64/pr94697.c
@@ -1,7 +1,7 @@
 /* { dg-do compile } */
 /* { dg-require-effective-target lp64 } */
 /* { dg-options "-O2 -mbranch-protection=standard" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 void bar (int *);
 void *addr;
diff --git gcc/testsuite/gcc.target/aarch64/torture/simd-abi-9.c gcc/testsuite/gcc.target/aarch64/torture/simd-abi-9.c
index 1119d004b3e..1205a6082fc 100644
--- gcc/testsuite/gcc.target/aarch64/torture/simd-abi-9.c
+++ gcc/testsuite/gcc.target/aarch64/torture/simd-abi-9.c
@@ -1,7 +1,7 @@
 /* { dg-do compile } */
 /* { dg-options "-fshrink-wrap -ffat-lto-objects" } */
 /* { dg-skip-if "" { *-*-* } { "-O0" } { "" } } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 int callee (void);
 
diff --git gcc/testsuite/gcc.target/aarch64/vect_mixed_sizes_14.c gcc/testsuite/gcc.target/aarch64/vect_mixed_sizes_14.c
index 00c90e4efdb..da121ceb39a 100644
--- gcc/testsuite/gcc.target/aarch64/vect_mixed_sizes_14.c
+++ gcc/testsuite/gcc.target/aarch64/vect_mixed_sizes_14.c
@@ -1,5 +1,5 @@
 /* { dg-options "-O2 -ftree-vectorize" } */
-/* { dg-final { check-function-bodies "**" "" } } */
+/* { dg-final { check-function-bodies "**" "" "" { target { ! aarch64*-*-darwin* } } } } */
 
 /*
 ** foo:
-- 
2.40.1

